service/src/main/java/bio/terra/pipelines/app/controller/PipelineRunsApiController.java

```diff
@@ -262,21 +260,19 @@ private ApiAsyncPipelineRunResponse pipelineRunToApi(PipelineRun pipelineRun, Pi
 
   /**
    * Returns the result endpoint corresponding to an async request. The endpoint is used to build an
-   * ApiJobReport. This method retrieves the protocol and domain name from the request and generates
-   * a result endpoint with the form: {protocol}{domainName}/{servletpath}/result/{jobId} relative
-   * to the async endpoint.
+   * ApiJobReport. This method retrieves the protocol and domain name from the ingress configuration
+   * and generates a result endpoint with the form:
+   * {protocol}{domainName}/api/pipelineruns/v1/result/{jobId}.
    *
-   * @param ingressConfiguration configuration specifying the ingress domain name
+   * @param domainName the domain name from the ingress configuration
    * @param jobId identifier for the job
    * @return a string with the result endpoint URL
    */
-  public static String getAsyncResultEndpoint(
-      IngressConfiguration ingressConfiguration, HttpServletRequest request, UUID jobId) {
-    String endpointPath = "%s/result/%s".formatted(request.getServletPath(), jobId);
+  public static String getAsyncResultEndpoint(String domainName, UUID jobId) {
```

Comment:
especially since this can be static, I'd advocate for moving this to somewhere else (JobApiUtils would be fine by me but wherever you think makes sense) since it's used by multiple controllers, and it's really needed because we're dealing with async Stairway jobs that require us to provide the result path as part of the JobReport (i.e. it's not something inherent to PipelineRuns).